packages/widgets/Source/Viewer/Viewer.js

```diff
@@ -1887,24 +1754,11 @@ Viewer.prototype._dataSourceRemoved = function (
 Viewer.prototype._onTick = function (clock) {
   const time = clock.currentTime;
 
-  const isUpdated = this._dataSourceDisplay.update(time);
-  if (this._allowDataSourcesToSuspendAnimation) {
+  const isUpdated = this._cesiumWidget.dataSourceDisplay.ready;
```

Comment:
I don't know if you noticed this change, but I'm not quite confident with it, and I have no idea how to do it differently. What do you think?

Reply:
I do remember thinking something about this seemed wrong but I forgot to loop back to it. I think it might be the case that this can be skipped because the `onTick` for the `CesiumWidget` checks the same thing and stops the same clock object if the data source display doesn't update? I also don't 100% understand what these functions are doing though so I might be wrong. Maybe we should even be initializing the `clock` differently in the `Viewer` now?

@ggetz would you be able to take a look specifically at this chunk of code/functionality?

Reply:
From what I understand, this block's intent is to temporarily stop the animation if something is still buffering into the dataSource. Since `ClockViewModel` only pass through the `Clock`'s `canAnimate` property, and the clock is already updated in `CesiumWidget`, I first thought that this was no more needed into the Viewer.
To be sure, I did try to duplicate the test "suspends animation by dataSources if allowed" from `CesiumWidget` spec to `Viewer` spec (replacing widget.clock by viewer.clockViewModel) and removed the mentioned code block: the test failed!
I guess this is because the `ClockViewModel` properties are not synchronized with `Clock`'s properties unless `ClockViewModel.synchronize()` is called. Therefore, in the case where `dataSource.update()` would return false, you would end up with `clock.canAnimate` set to false, and `clockViewModel.canAnimate` still equal to true.
May be the clockViewModel is being synchronized later but I can't be certain...

Reply:
Yes, I think this is a sensible change. Just as @jfayot stated, we need to update the `clockViewModel`, which `CesiumWidget` does not have a reference to.